src/Nethermind/Nethermind.Core/BlockToProduce.cs

```diff
@@ -49,5 +49,10 @@ public BlockToProduce(BlockHeader blockHeader, IEnumerable<Transaction> transact
         public BlockToProduce(BlockHeader blockHeader) : base(blockHeader)
         {
         }
+
+        public override Block CreateBlockForProcessing(BlockHeader header) => 
+            new BlockToProduce(header, Transactions, Ommers);
```

Comment:
remove from here, move to blockchain namespaces, remove methods from block and keep in a separate class